docs/madr/decisions/075-migrating-to-kri-based-envoy-resource-and-stat-naming.md

```diff
@@ -0,0 +1,167 @@
+
+* Log or show a warning if a `Dataplane` is matched by a `MeshProxyPatch`, but none of the patch modifiers apply to any xDS resources
+
+  *Problem:* It's hard to determine whether complex patches like JSON Patch match anything. There's no existing way to surface this to the GUI, and logging such cases could be noisy and lead to false positives. Even in debug logs, it could quickly become more annoying than helpful.
+
+* Search for the `kri_` prefix in patch contents as a hint that the policy is KRI-aware
+
+  *Problem:* This is unreliable, and all the concerns above still apply.
+
+* Show a general warning in the dataplane or policy view when KRI is enabled and any `MeshProxyPatch` applies
+
+  *Problem:* This might be useful the first time but would quickly become noise once the user verifies their configuration.
+
+Because none of these options are satisfactory, the chosen path is to document this behavior clearly:
+
+* Include a warning in the upgrade notes to alert users that existing `MeshProxyPatch` policies must be reviewed and updated before enabling KRI naming
+* Add a strong warning to the documentation for `MeshProxyPatch` policies
+
+### Updating ZoneIgress and ZoneEgress overview resources with feature flags
+
+To support the Kuma GUI in adapting to KRI-based naming, we need to expose feature flag information in `ZoneIngressOverview` and `ZoneEgressOverview` resources, similar to how it's already done for `DataplaneOverview`. These overview resources are available via the control plane API and provide a summary of runtime state and metadata.
+
+We will update both `ZoneIngressOverview` and `ZoneEgressOverview` to include metadata with active feature flags. This will allow the GUI to detect whether KRI naming is enabled for each proxy and adjust its behavior accordingly.
+
+## Migration
+
+### Kuma GUI
+
+The Kuma GUI relies on Envoy stat names to display inbound and outbound endpoint details for data plane, `ZoneIngress`, and `ZoneEgress` proxies. These names are used to associate metrics with specific resources and visualize traffic paths in the GUI.
+
+To support the KRI naming format, we need to update the GUI parsers that extract resource information from stat names such as listeners, clusters, and endpoints. These parsers must recognize and handle the new KRI format defined in the [Resource Identifier MADR](https://github.com/kumahq/kuma/blob/d19b78a4556962f4d9d3cc5921c7bdc73dc93d26/docs/madr/decisions/070-resource-identifier.md?plain=1#L328):
+
+```
+kri_<resource-type>_<mesh>_<zone>_<namespace>_<resource-name>_<section-name>
+```
+
+The updated parsers will be activated when the `feature-kri-naming` flag is present in the metadata of the `DataplaneOverview`, `ZoneIngressOverview`, and `ZoneEgressOverview` resources.
+
+The GUI does not rely on the names of xDS resources themselves. It only uses stat names, so no additional changes are required to support KRI naming.
```

Comment:
Just wanted to point out the only bit in the code where we don't just "simply render" XDS config:

https://github.com/kumahq/kuma-gui/blob/d2548eea3bdbc1787b574aee69b3da453d7eb743/packages/kuma-gui/src/app/connections/sources.ts#L189-L203

This is for when we show the XDS config in the side drawer once you've clicked on an inbound/outbound card. We pick out the bits that we should show for a specific inbound/outbound.

I don't think it's super important, but wanted to mention incase, and I think from our perspective its straightforwards, we will just have a fork here this based on the feature flag if necessary.